# Nama: Taqiyuddin Ja'far
## NIM: 250401020186
### Kelas: IF405

In [1]:
import pandas as pd
import numpy as np

#Muat dataset titanic
sumber = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(sumber)
df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [2]:
# Eksplorasi Awal
# Q1: Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?
print(f"Jumlah baris dan kolom dalam format (baris,kolom): {df.shape}") # shape menjawab (baris,kolom)
print(f"Daftar kolom: {df.columns.tolist()}\n") #columns.tolist menampilkan daftar 12 kolom mulai dari 'PassengerId' sampai 'Embarked'


Jumlah baris dan kolom dalam format (baris,kolom): (891, 12)
Daftar kolom: ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']



In [3]:
# Q2: Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?
missing_data = df.isnull().sum()
print(missing_data) # pemeriksaan missing values. Jadi di kolom 'Age' ada 177, di 'Cabin' ada 687, di 'Embarked' ada 2
total_rows = len(df)
print(f"total penumpang = {total_rows}\n") # len(df) menghitung jumlah baris atau dalam data ini sebagai total populasi semua penumpang

missing_info = pd.DataFrame({
    'Jumlah Missing': missing_data,
    'Proporsi (%)': (missing_data / total_rows) * 100
})
print(missing_info[missing_info['Jumlah Missing'] > 0], "\n")

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
total penumpang = 891

          Jumlah Missing  Proporsi (%)
Age                  177     19.865320
Cabin                687     77.104377
Embarked               2      0.224467 



In [4]:
# Q3: Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas.
percent_survived = df['Survived'].mean() * 100 # Menggunakan Pandas mean() dari seluruh kolom 'Survived',karena kolom 'Survived' berisi 0 dan 1
print(f"Persentase penumpang selamat: {percent_survived:.2f}%\n")

Persentase penumpang selamat: 38.38%



In [5]:
# Filtering — Penumpang Wanita Kelas 1
# Q4: Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?
female_class_1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)] # Menggunakan Boolean Indexing dengan kondisi ganda
print(f"Jumlah penumpang wanita dari kelas 1: {len(female_class_1)}")
print(f"Rata-rata usia penumpang wanita dari kelas 1: {female_class_1['Age'].mean():.2f} tahun\n") # Menggunakan Pandas mean() dari seluruh kolom 'Age' dari female_class_1

Jumlah penumpang wanita dari kelas 1: 94
Rata-rata usia penumpang wanita dari kelas 1: 34.61 tahun



In [6]:
# Groupby & Agregasi
# Q5: Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?
survival_by_class = df.groupby('Pclass')['Survived'].mean()
print(survival_by_class)
print("\n")

hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean'),
).round(2)
print(hasil) # hasil aggregasi
print("\n")

print(f"Kelas dengan tingkat keselamatan tertinggi: {survival_by_class.idxmax()}\n")

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


        total  selamat
Pclass                
1         216     0.63
2         184     0.47
3         491     0.24


Kelas dengan tingkat keselamatan tertinggi: 1



In [7]:
# Q6: Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.

# Analisis tingkat keselamatan berdasarkan pelabuhan keberangkatan (Embarked)
# C = Cherbourg, Q = Queenstown, S = Southampton

embarked_survival = df.groupby('Embarked')['Survived'].agg(['count', 'mean']).round(2).sort_values(by='mean', ascending=False)

print("Analisis Berdasarkan Pelabuhan Keberangkatan:")
print(embarked_survival)

# Interpretasi singkat:
# Penumpang yang naik dari Cherbourg (C) memiliki tingkat keselamatan tertinggi (sekitar 55%),
# sementara dari Southampton (S) memiliki tingkat keselamatan terendah meskipun jumlah penumpangnya paling banyak.

Analisis Berdasarkan Pelabuhan Keberangkatan:
          count  mean
Embarked             
C           168  0.55
Q            77  0.39
S           644  0.34


## Kesimpulan

Berdasarkan analisis yang telah saya lakukan terhadap dataset Titanic, berikut adalah rangkuman dari apa yang telah saya pelajari:

**Apa yang Saya Pelajari:**
Saya belajar cara melakukan eksplorasi data dasar menggunakan library Pandas dan NumPy, mulai dari memuat data dari sumber eksternal, memeriksa struktur dataset (baris, kolom, dan tipe data), hingga mengidentifikasi data yang hilang (*missing values*).

**Temuan Utama:**
1. **Ketimpangan Keselamatan:** Saya menemukan bahwa kelas sosial berpengaruh besar terhadap peluang bertahan hidup; penumpang Kelas 1 memiliki tingkat keselamatan tertinggi (63%), sementara Kelas 3 hanya 24%.
2. **Profil Penumpang:** Saya berhasil menyaring data khusus untuk penumpang wanita Kelas 1 yang berjumlah 94 orang dengan rata-rata usia sekitar 34,6 tahun.
3. **Faktor Geografis:** Saya juga menemukan bahwa pelabuhan keberangkatan Cherbourg memiliki rata-rata keselamatan lebih tinggi (55%) dibandingkan pelabuhan lainnya.

**Keterbatasan dan Pertanyaan yang Muncul:**
* **Data yang Hilang:** Saya menyadari adanya keterbatasan data yang besar pada kolom 'Cabin' (77% hilang) dan 'Age' (20% hilang), yang membuat analisis usia menjadi kurang akurat jika tidak ditangani lebih lanjut.
* **Pertanyaan:** Saya jadi terpikir bagaimana mengatasi data-data yang hilang ini selanjutnya